src/app/CommandResponseHelper.h

```diff
@@ -40,9 +41,22 @@ class CommandResponseHelper
         return CHIP_NO_ERROR;
     };
 
+    // WARNING: Do not use for basic response-payload-free success, as SUCCESS status
+    // code should be handled without a cluster-specific status.
     CHIP_ERROR Success(ClusterStatus aClusterStatus)
     {
-        CHIP_ERROR err = mCommandHandler->AddClusterSpecificSuccess(mCommandPath, aClusterStatus);
+        CHIP_ERROR err = mCommandHandler->FallibleAddStatus(
+            mCommandPath, Protocols::InteractionModel::ClusterStatusCode::ClusterSpecificSuccess(aClusterStatus));
+        if (err == CHIP_NO_ERROR)
+        {
+            mSentResponse = true;
+        }
```

Comment:
I wonder why this was not `mSentResponse = (err == CHIP_NO_ERROR);`? Does it assume mSentResponse may have been true already and we do not want to change that?

Do you happen to know the existing logic reason?